sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -935,9 +941,12 @@ ur_program_handle_t ProgramManager::getBuiltURProgram(
     CacheKey.first.second = BImg->getImageID();
     bool DidInsert = Cache.insertBuiltProgram(CacheKey, ResProgram);
     if (DidInsert) {
+      // Add to the eviction list.
+      Cache.registerProgramFetch(CacheKey, ResProgram, 1);
       // For every cached copy of the program, we need to increment its refcount
       Adapter->call<UrApiKind::urProgramRetain>(ResProgram);
-    }
+    } else
+      Cache.registerProgramFetch(CacheKey, ResProgram, 0);
```

Comment:
Nit; Seems like we could just do `Cache.registerProgramFetch(CacheKey, ResProgram, DidInsert);` before the conditional.